query-engine/connectors/sql-query-connector/src/query_builder/select/lateral.rs

```diff
@@ -6,13 +6,22 @@ use crate::{
     model_extensions::AsColumn,
 };
 
+use std::collections::HashMap;
+
 use quaint::ast::*;
 use query_structure::*;
 
 /// Select builder for joined queries. Relations are resolved using LATERAL JOINs.
 #[derive(Debug, Default)]
 pub(crate) struct LateralJoinSelectBuilder {
     alias: Alias,
+
+    // TODO: consider using references to `VirtualSelection` as keys to avoid cloning. This would
```

Comment:
Alternatively, we can still have owned but more lightweight data: we don't need filters here, and it's the most heavyweight part of `VirtualSelection`.

Reply:
will change this tomorrow